crates/common/light_client/src/update.rs

```diff
@@ -0,0 +1,122 @@
+use alloy_primitives::B256;
+use anyhow::ensure;
+use ream_consensus::{
+    constants::{GENESIS_SLOT, MIN_SYNC_COMMITTEE_PARTICIPANTS},
+    electra::{beacon_block::SignedBeaconBlock, beacon_state::BeaconState},
+    misc::compute_sync_committee_period_at_slot,
+    sync_aggregate::SyncAggregate,
+    sync_committee::SyncCommittee,
+};
+use serde::{Deserialize, Serialize};
+use ssz_derive::{Decode, Encode};
+use ssz_types::{FixedVector, typenum::U6};
+use tree_hash::TreeHash;
+use tree_hash_derive::TreeHash;
+
+use crate::header::LightClientHeader;
+
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize, Encode, Decode, TreeHash)]
+pub struct LightClientUpdate {
+    /// Header attested to by the sync committee
+    pub attested_header: LightClientHeader,
+    /// Next sync committee corresponding to `attested_header.beacon.state_root`
+    pub next_sync_committee: SyncCommittee,
+    pub next_sync_committee_branch: FixedVector<B256, U6>,
+    /// Finalized header corresponding to `attested_header.beacon.state_root`
+    pub finalized_header: LightClientHeader,
+    pub finality_branch: FixedVector<B256, U6>,
+    /// Sync committee aggregate signature
+    pub sync_aggregate: SyncAggregate,
+    /// Slot at which the aggregate signature was created (untrusted)
+    #[serde(with = "serde_utils::quoted_u64")]
+    pub signature_slot: u64,
+}
+
+impl LightClientUpdate {
+    pub fn new(
+        state: BeaconState,
+        block: SignedBeaconBlock,
+        attested_state: BeaconState,
+        attested_block: SignedBeaconBlock,
+        finalized_block: Option<SignedBeaconBlock>,
+    ) -> anyhow::Result<Self> {
+        ensure!(
+            block
+                .message
+                .body
+                .sync_aggregate
+                .sync_committee_bits
+                .iter()
+                .filter(|sync_committee_bit| *sync_committee_bit)
+                .count()
+                >= MIN_SYNC_COMMITTEE_PARTICIPANTS.try_into().unwrap(),
+            "Not enough sync committee participants"
+        );
```

Comment:
```suggestion
        ensure!(
            block
                .message
                .body
                .sync_aggregate
                .sync_committee_bits
                .iter()
                .filter(|sync_committee_bit| *sync_committee_bit)
                .count() as u64
                >= MIN_SYNC_COMMITTEE_PARTICIPANTS,
            "Not enough sync committee participants"
        );
```

don't use unwrap in production code